controllers/authorino_controller.go

```diff
@@ -443,73 +478,96 @@ func (r *AuthorinoReconciler) createAuthorinoPermission(authorino *api.Authorino
+		return fmt.Sprintf("%s-%s", crName, roleBindingNameSuffix)
+	}
+}
+
+func (r *AuthorinoReconciler) reconcileAuthorinoRoleBinding(roleBindingNameSuffix, roleName string, serviceAccount k8score.ServiceAccount, clusterScoped bool, authorino *api.Authorino) error {
+	var logger = r.Log
+
+	var roleBinding client.Object
+	var roleBindingName = authorinoRoleBindingName(authorino.Name, roleBindingNameSuffix, clusterScoped)
+	var roleBindingSearchName types.NamespacedName
+
+	if clusterScoped {
+		roleBinding = authorinoResources.GetAuthorinoClusterRoleBinding(roleName, serviceAccount)
+		roleBindingSearchName = types.NamespacedName{Name: roleBindingName}
+	} else {
+		roleBinding = authorinoResources.GetAuthorinoRoleBinding(roleName, serviceAccount)
+		roleBinding.SetNamespace(authorino.Namespace)
+		roleBindingSearchName = types.NamespacedName{Name: roleBindingName, Namespace: authorino.Namespace}
 	}
-	bindingName := resourcePrefixName + "-authorino"
-	bindingNsdName := namespacedName(authorino.Namespace, bindingName)
-	if err := r.Get(context.TODO(), bindingNsdName, binding); err != nil {
+
+	if err := r.Get(context.TODO(), roleBindingSearchName, roleBinding); err != nil {
+		// failed to get (cluster)rolebinding -> check if not found or other error
+
 		if errors.IsNotFound(err) {
-			ctrl.SetControllerReference(authorino, binding, r.Scheme)
-			// doesn't exist - create one
-			binding.SetName(bindingName)
-			if err := r.Client.Create(context.TODO(), binding); err != nil {
+			// (cluster)rolebinding does not exist -> create one
+			roleBinding.SetName(roleBindingName)
+			_ = ctrl.SetControllerReference(authorino, roleBinding, r.Scheme) // useful for namespaced role bindings
+			if err := r.Client.Create(context.TODO(), roleBinding); err != nil {
 				return r.wrapErrorWithStatusUpdate(
 					logger, authorino, r.setStatusFailed(api.AuthorinoUnableToCreateBindingForClusterRole),
-					fmt.Errorf("failed to create %s binding for authorino ClusterRole, err: %v", bindingName, err),
+					fmt.Errorf("failed to create %s binding for authorino ClusterRole, err: %v", roleBindingName, err),
 				)
 			}
 		}
+
+		// other error -> return
 		return r.wrapErrorWithStatusUpdate(
 			logger, authorino, r.setStatusFailed(api.AuthorinoUnableToGetBindingForClusterRole),
-			fmt.Errorf("failed to get %s binding for authorino ClusterRole, err: %v", bindingName, err),
+			fmt.Errorf("failed to get %s binding for authorino ClusterRole, err: %v", roleBindingName, err),
 		)
-	}
+	} else {
+		// (cluster)rolebinding exists -> ensure it includes the serviceaccount among the subjects
 
-	// creates leader election role
-	// TODO: describe what this leader election is
-	return r.leaderElectionPermission(authorino, sa.GetName())
+		rb := authorinoResources.AppendSubjectToRoleBinding(roleBinding, serviceAccount) // TODO: Remove subject from the clusterrolebinding when the authorino instance is deleted. OwnerReference will not work for cluster-scoped resources :/
```

Comment:
Nice one :)

Reply:
And it's idempotent...and indifferent to `RoleBinding` or `ClusterRoleBinding` 😊